# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsChild.cpp
# Commit: 9e7a90dcc429
# Full Hash: 9e7a90dcc429194fbe08e63aeed311ce100114cc
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-10-02 03:38:52
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Use const where easily possible. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43631
# ==============================================================================

diff -r e493a6b5f125 -r 9e7a90dcc429 dom/indexedDB/ActorsChild.cpp
--- a/dom/indexedDB/ActorsChild.cpp	Tue Oct 01 08:56:37 2019 +0000
+++ b/dom/indexedDB/ActorsChild.cpp	Tue Oct 01 09:14:04 2019 +0000
@@ -189,7 +189,7 @@
 };
 
 class MOZ_STACK_CLASS ResultHelper final : public IDBRequest::ResultCallback {
-  IDBRequest* mRequest;
+  IDBRequest* const mRequest;
   AutoSetCurrentTransaction mAutoTransaction;
 
   union {
@@ -372,7 +372,7 @@
       return NS_OK;
     }
 
-    bool ok = GetOrCreateDOMReflector(aCx, aDOMObject, aResult);
+    const bool ok = GetOrCreateDOMReflector(aCx, aDOMObject, aResult);
     if (NS_WARN_IF(!ok)) {
       IDB_REPORT_INTERNAL_ERR();
       return NS_ERROR_DOM_INDEXEDDB_UNKNOWN_ERR;
@@ -383,7 +383,7 @@
 
   nsresult GetResult(JSContext* aCx, StructuredCloneReadInfo* aCloneInfo,
                      JS::MutableHandle<JS::Value> aResult) {
-    bool ok = IDBObjectStore::DeserializeValue(aCx, *aCloneInfo, aResult);
+    const bool ok = IDBObjectStore::DeserializeValue(aCx, *aCloneInfo, aResult);
 
     if (NS_WARN_IF(!ok)) {
       return NS_ERROR_DOM_DATA_CLONE_ERR;
@@ -415,7 +415,7 @@
 
         JS::Rooted<JS::Value> value(aCx);
 
-        nsresult rv = GetResult(aCx, &cloneInfo, &value);
+        const nsresult rv = GetResult(aCx, &cloneInfo, &value);
         if (NS_WARN_IF(NS_FAILED(rv))) {
           return rv;
         }
@@ -434,7 +434,7 @@
 
   nsresult GetResult(JSContext* aCx, const Key* aKey,
                      JS::MutableHandle<JS::Value> aResult) {
-    nsresult rv = aKey->ToJSVal(aCx, aResult);
+    const nsresult rv = aKey->ToJSVal(aCx, aResult);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -463,7 +463,7 @@
 
         JS::Rooted<JS::Value> value(aCx);
 
-        nsresult rv = GetResult(aCx, &key, &value);
+        const nsresult rv = GetResult(aCx, &key, &value);
         if (NS_WARN_IF(NS_FAILED(rv))) {
           return rv;
         }
@@ -499,7 +499,7 @@
   }
 
  protected:
-  ~PermissionRequestMainProcessHelper() {}
+  ~PermissionRequestMainProcessHelper() = default;
 
  private:
   virtual void OnPromptComplete(PermissionValue aPermissionValue) override;
@@ -531,13 +531,13 @@
 
           const IPCBlob& ipcBlob = blobOrMutableFile.get_IPCBlob();
 
-          RefPtr<BlobImpl> blobImpl = IPCBlobUtils::Deserialize(ipcBlob);
+          const RefPtr<BlobImpl> blobImpl = IPCBlobUtils::Deserialize(ipcBlob);
           MOZ_ASSERT(blobImpl);
 
           RefPtr<Blob> blob =
               Blob::Create(aDatabase->GetOwnerGlobal(), blobImpl);
 
-          StructuredCloneFile* file = aFiles.AppendElement();
+          StructuredCloneFile* const file = aFiles.AppendElement();
           MOZ_ASSERT(file);
 
           file->mType = StructuredCloneFile::eBlob;
@@ -553,7 +553,7 @@
 
           switch (blobOrMutableFile.type()) {
             case BlobOrMutableFile::Tnull_t: {
-              StructuredCloneFile* file = aFiles.AppendElement();
+              StructuredCloneFile* const file = aFiles.AppendElement();
               MOZ_ASSERT(file);
 
               file->mType = StructuredCloneFile::eMutableFile;
@@ -562,17 +562,17 @@
             }
 
             case BlobOrMutableFile::TPBackgroundMutableFileChild: {
-              auto* actor = static_cast<BackgroundMutableFileChild*>(
+              auto* const actor = static_cast<BackgroundMutableFileChild*>(
                   blobOrMutableFile.get_PBackgroundMutableFileChild());
               MOZ_ASSERT(actor);
 
               actor->EnsureDOMObject();
 
-              auto* mutableFile =
+              auto* const mutableFile =
                   static_cast<IDBMutableFile*>(actor->GetDOMObject());
               MOZ_ASSERT(mutableFile);
 
-              StructuredCloneFile* file = aFiles.AppendElement();
+              StructuredCloneFile* const file = aFiles.AppendElement();
               MOZ_ASSERT(file);
 
               file->mType = StructuredCloneFile::eMutableFile;
@@ -596,13 +596,14 @@
 
             const IPCBlob& ipcBlob = blobOrMutableFile.get_IPCBlob();
 
-            RefPtr<BlobImpl> blobImpl = IPCBlobUtils::Deserialize(ipcBlob);
+            const RefPtr<BlobImpl> blobImpl =
+                IPCBlobUtils::Deserialize(ipcBlob);
             MOZ_ASSERT(blobImpl);
 
             RefPtr<Blob> blob =
                 Blob::Create(aDatabase->GetOwnerGlobal(), blobImpl);
 
-            StructuredCloneFile* file = aFiles.AppendElement();
+            StructuredCloneFile* const file = aFiles.AppendElement();
             MOZ_ASSERT(file);
 
             file->mType = StructuredCloneFile::eStructuredClone;
@@ -610,7 +611,7 @@
           } else {
             MOZ_ASSERT(blobOrMutableFile.type() == BlobOrMutableFile::Tnull_t);
 
-            StructuredCloneFile* file = aFiles.AppendElement();
+            StructuredCloneFile* const file = aFiles.AppendElement();
             MOZ_ASSERT(file);
 
             file->mType = StructuredCloneFile::eStructuredClone;
@@ -623,7 +624,7 @@
         case StructuredCloneFile::eWasmCompiled: {
           MOZ_ASSERT(blobOrMutableFile.type() == BlobOrMutableFile::Tnull_t);
 
-          StructuredCloneFile* file = aFiles.AppendElement();
+          StructuredCloneFile* const file = aFiles.AppendElement();
           MOZ_ASSERT(file);
 
           file->mType = serializedFile.type();
@@ -653,8 +654,8 @@
 
   AUTO_PROFILER_LABEL("IndexedDB:DispatchErrorEvent", DOM);
 
-  RefPtr<IDBRequest> request = aRequest;
-  RefPtr<IDBTransaction> transaction = aTransaction;
+  const RefPtr<IDBRequest> request = aRequest;
+  const RefPtr<IDBTransaction> transaction = aTransaction;
 
   request->SetError(aErrorCode);
 
@@ -686,7 +687,8 @@
   }
 
   IgnoredErrorResult rv;
-  bool doDefault = request->DispatchEvent(*aEvent, CallerType::System, rv);
+  const bool doDefault =
+      request->DispatchEvent(*aEvent, CallerType::System, rv);
   if (NS_WARN_IF(rv.Failed())) {
     return;
   }
@@ -699,7 +701,7 @@
   // later.
   if (transaction && transaction->IsOpen() &&
       aErrorCode != NS_ERROR_DOM_INDEXEDDB_ABORT_ERR) {
-    WidgetEvent* internalEvent = aEvent->WidgetEventPtr();
+    WidgetEvent* const internalEvent = aEvent->WidgetEventPtr();
     MOZ_ASSERT(internalEvent);
 
     if (internalEvent->mFlags.mExceptionWasRaised) {
@@ -716,11 +718,11 @@
 
   AUTO_PROFILER_LABEL("IndexedDB:DispatchSuccessEvent", DOM);
 
-  RefPtr<IDBRequest> request = aResultHelper->Request();
+  const RefPtr<IDBRequest> request = aResultHelper->Request();
   MOZ_ASSERT(request);
   request->AssertIsOnOwningThread();
 
-  RefPtr<IDBTransaction> transaction = aResultHelper->Transaction();
+  const RefPtr<IDBTransaction> transaction = aResultHelper->Transaction();
 
   if (transaction && transaction->IsAborted()) {
     DispatchErrorEvent(request, transaction->AbortCode(), transaction);
@@ -762,7 +764,7 @@
     return;
   }
 
-  WidgetEvent* internalEvent = aEvent->WidgetEventPtr();
+  WidgetEvent* const internalEvent = aEvent->WidgetEventPtr();
   MOZ_ASSERT(internalEvent);
 
   if (transaction && transaction->IsOpen()) {
@@ -778,13 +780,13 @@
 PRFileDesc* GetFileDescriptorFromStream(nsIInputStream* aStream) {
   MOZ_ASSERT(aStream);
 
-  nsCOMPtr<nsIFileMetadata> fileMetadata = do_QueryInterface(aStream);
+  const nsCOMPtr<nsIFileMetadata> fileMetadata = do_QueryInterface(aStream);
   if (NS_WARN_IF(!fileMetadata)) {
     return nullptr;
   }
 
   PRFileDesc* fileDesc;
-  nsresult rv = fileMetadata->GetFileDescriptor(&fileDesc);
+  const nsresult rv = fileMetadata->GetFileDescriptor(&fileDesc);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return nullptr;
   }
@@ -865,7 +867,7 @@
 
   NS_IMETHOD
   Run() override {
-    bool completed = RunInternal();
+    const bool completed = RunInternal();
     if (completed) {
       OperationCompleted();
     }
@@ -875,7 +877,7 @@
 
   void OperationCompleted() {
     if (NS_IsMainThread()) {
-      RefPtr<WorkerPermissionOperationCompleted> runnable =
+      const RefPtr<WorkerPermissionOperationCompleted> runnable =
           new WorkerPermissionOperationCompleted(mWorkerPrivate, this);
 
       MOZ_ALWAYS_TRUE(runnable->Dispatch());
@@ -907,20 +909,20 @@
       wp = wp->GetParent();
     }
 
-    nsPIDOMWindowInner* window = wp->GetWindow();
+    nsPIDOMWindowInner* const window = wp->GetWindow();
     if (!window) {
       return true;
     }
 
     nsresult rv;
-    nsCOMPtr<nsIPrincipal> principal =
+    const nsCOMPtr<nsIPrincipal> principal =
         mozilla::ipc::PrincipalInfoToPrincipal(mPrincipalInfo, &rv);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return true;
     }
 
     if (XRE_IsParentProcess()) {
-      nsCOMPtr<Element> ownerElement =
+      const nsCOMPtr<Element> ownerElement =
           do_QueryInterface(window->GetChromeEventHandler());
       if (NS_WARN_IF(!ownerElement)) {
         return true;
@@ -956,10 +958,10 @@
   }
 
  private:
-  WorkerPrivate* mWorkerPrivate;
+  WorkerPrivate* const mWorkerPrivate;
   BackgroundFactoryRequestChild* mActor;
   RefPtr<IDBFactory> mFactory;
-  PrincipalInfo mPrincipalInfo;
+  const PrincipalInfo mPrincipalInfo;
 };
 
 void WorkerPermissionRequest::OnPromptComplete(
@@ -1018,7 +1020,7 @@
 
 class MOZ_STACK_CLASS FileHandleResultHelper final
     : public IDBFileRequest::ResultCallback {
-  IDBFileRequest* mFileRequest;
+  IDBFileRequest* const mFileRequest;
   AutoSetCurrentFileHandle mAutoFileHandle;
 
   union {
@@ -1119,7 +1121,7 @@
  private:
   nsresult GetResult(JSContext* aCx, File* aFile,
                      JS::MutableHandle<JS::Value> aResult) {
-    bool ok = GetOrCreateDOMReflector(aCx, aFile, aResult);
+    const bool ok = GetOrCreateDOMReflector(aCx, aFile, aResult);
     if (NS_WARN_IF(!ok)) {
       return NS_ERROR_DOM_FILEHANDLE_UNKNOWN_ERR;
     }
@@ -1203,7 +1205,7 @@
 
 already_AddRefed<File> ConvertActorToFile(
     IDBFileHandle* aFileHandle, const FileRequestGetFileResponse& aResponse) {
-  auto* actor = static_cast<PendingIPCBlobChild*>(aResponse.fileChild());
+  auto* const actor = static_cast<PendingIPCBlobChild*>(aResponse.fileChild());
 
   IDBMutableFile* mutableFile = aFileHandle->GetMutableFile();
   MOZ_ASSERT(mutableFile);
@@ -1216,12 +1218,12 @@
   const Maybe<int64_t>& lastModified = metadata.lastModified();
   MOZ_ASSERT(lastModified.isSome());
 
-  RefPtr<BlobImpl> blobImpl = actor->SetPendingInfoAndDeleteActor(
+  const RefPtr<BlobImpl> blobImpl = actor->SetPendingInfoAndDeleteActor(
       mutableFile->Name(), mutableFile->Type(), size.value(),
       lastModified.value());
   MOZ_ASSERT(blobImpl);
 
-  RefPtr<BlobImpl> blobImplSnapshot =
+  const RefPtr<BlobImpl> blobImplSnapshot =
       new BlobImplSnapshot(blobImpl, static_cast<IDBFileHandle*>(aFileHandle));
 
   RefPtr<File> file =
@@ -1238,8 +1240,8 @@
   MOZ_ASSERT(NS_ERROR_GET_MODULE(aErrorCode) == NS_ERROR_MODULE_DOM_FILEHANDLE);
   MOZ_ASSERT(aFileHandle);
 
-  RefPtr<IDBFileRequest> fileRequest = aFileRequest;
-  RefPtr<IDBFileHandle> fileHandle = aFileHandle;
+  const RefPtr<IDBFileRequest> fileRequest = aFileRequest;
+  const RefPtr<IDBFileHandle> fileHandle = aFileHandle;
 
   AutoSetCurrentFileHandle ascfh(aFileHandle);
 
@@ -1251,11 +1253,11 @@
 void DispatchFileHandleSuccessEvent(FileHandleResultHelper* aResultHelper) {
   MOZ_ASSERT(aResultHelper);
 
-  RefPtr<IDBFileRequest> fileRequest = aResultHelper->FileRequest();
+  const RefPtr<IDBFileRequest> fileRequest = aResultHelper->FileRequest();
   MOZ_ASSERT(fileRequest);
   fileRequest->AssertIsOnOwningThread();
 
-  RefPtr<IDBFileHandle> fileHandle = aResultHelper->FileHandle();
+  const RefPtr<IDBFileHandle> fileHandle = aResultHelper->FileHandle();
   MOZ_ASSERT(fileHandle);
 
   if (fileHandle->IsAborted()) {
@@ -1298,13 +1300,13 @@
     Completed
   };
 
-  nsCOMPtr<nsIEventTarget> mOwningEventTarget;
+  const nsCOMPtr<nsIEventTarget> mOwningEventTarget;
   RefPtr<TaskQueue> mTaskQueue;
   nsCOMPtr<nsIEventTarget> mTaskQueueEventTarget;
   nsCOMPtr<nsIInputStream> mStream;
   UniquePtr<JSStructuredCloneData> mCloneData;
   BackgroundRequestChild* mActor;
-  uint32_t mCloneDataIndex;
+  const uint32_t mCloneDataIndex;
   nsresult mResultCode;
   State mState;
 
@@ -1469,7 +1471,7 @@
     const DatabaseSpec& aSpec, PBackgroundIDBFactoryRequestChild* aRequest) {
   AssertIsOnOwningThread();
 
-  auto request = static_cast<BackgroundFactoryRequestChild*>(aRequest);
+  auto* const request = static_cast<BackgroundFactoryRequestChild*>(aRequest);
   MOZ_ASSERT(request);
 
   return new BackgroundDatabaseChild(aSpec, request);